neqo-transport/src/connection/tests/datagram.rs

```diff
@@ -224,6 +222,48 @@ fn datagram_acked() {
     ));
 }
 
+#[test]
+fn datagram_after_stream_data() {
+    let (mut client, mut server) = connect_datagram();
+
+    // Wriate a datagram first.
+    let _dgram_sent = client.stats().frame_tx.datagram;
+    assert_eq!(client.send_datagram(DATA_SMALLER_THAN_MTU, Some(1)), Ok(()));
+
+    // Create a stream and send some data.
+    let stream_id = client.stream_create(StreamType::BiDi).unwrap();
+    client.stream_send(stream_id, &[6; 100]).unwrap();
```

Comment:
Ok, so is the datagram too big to fit in a packet that has this stream data as well?  I'd send a little more to ensure that is the case.

If it is, then that is good.  Then you want to:

1. have the client create two packets.  the first should be smaller than the datagram size, which you can check
2. have the server process the first.  it should receive stream data, but not a datagram event
3. have the server process the second.  it should receive the datagram event then

We probably shouldn't rely on the order of events here.  That's not something we'd want to promise in the API unless we really had to (and we don't).

Then add another test that sends low priority stream data first and observe the opposite inversion.

Reply:
Will do